addons/rmsmartshape/shapes/shape_base.gd

```diff
@@ -1450,6 +1398,21 @@ func _build_edge_with_material(
 	var c_scale := 1.0
 	var c_extends := 0.0
 
+	# Cache points index maps
+	# Map of all tesselated point index to its corresponding vertex index
+	var t_point_idx_to_point_idx: Array[int] = []
+	# Map of all vertex index to their corresponding tesselated points indices
+	var point_idx_to_t_points_idx: Array[Array] = []
+	var point_idx = -1
+	for t_point_idx in verts_t.size():
+		var next_point_idx = _get_next_point_index_wrap_around(point_idx, verts)
+		if verts_t[t_point_idx] == verts[next_point_idx]:
+			point_idx = next_point_idx
+			point_idx_to_t_points_idx.push_back([])
+		
+		t_point_idx_to_point_idx.push_back(point_idx)
+		point_idx_to_t_points_idx[point_idx].push_back(t_point_idx)
```

Comment:
Here is the index maps. I've put it naively at the start of the _build_edge_with_material, but it could be improved for readability.

Also, there are several identical calculations that are performed by each thread that does not depend on the index_map, including this one. The result is always the same, it would be better if it was calculated only once and the result was shared to all threads.